diff --git a/portfile.cmake b/portfile.cmake
index b43eedd..3b255bf 100644
--- a/portfile.cmake
+++ b/portfile.cmake
@@ -48,11 +48,15 @@ include("${cmake_vars_file}")
 
 # There are compilation errors on gcc 15. adding `-std=c17` to CFLAGS for workaround.
 # ref: https://gitlab.archlinux.org/archlinux/packaging/packages/ncurses/-/issues/3
-list(APPEND OPTIONS "CFLAGS=\$CFLAGS -std=c17")
+if(VCPKG_DETECTED_CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND VCPKG_DETECTED_CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 15)
+    set(ENV{CFLAGS} "$ENV{CFLAGS} -std=c17")
+endif()
 
-vcpkg_make_configure(
+vcpkg_configure_make(
     SOURCE_PATH "${SOURCE_PATH}"
-    DEFAULT_OPTIONS_EXCLUDE "^--docdir"
+    CONFIGURE_ENVIRONMENT_VARIABLES CFLAGS
+    DETERMINE_BUILD_TRIPLET
+    NO_ADDITIONAL_PATHS
     OPTIONS
         ${OPTIONS}
         --disable-db-install
@@ -60,6 +64,7 @@ vcpkg_make_configure(
         --disable-rpath-hack
         --enable-pc-files
         --without-ada
+        --without-cxx-binding
         --without-debug # "lib model"
         --without-manpages
         --without-progs
@@ -67,7 +72,7 @@ vcpkg_make_configure(
         --without-tests
         --with-pkg-config-libdir=libdir
 )
-vcpkg_make_install()
+vcpkg_install_make()
 vcpkg_fixup_pkgconfig()
 
 # Prefer local files over search path
@@ -76,14 +81,6 @@ foreach(file IN LISTS headers)
     vcpkg_replace_string("${file}" [[#include <ncursesw/([^>]*)>]] [[#include "\1"]] REGEX IGNORE_UNCHANGED)
 endforeach()
 
-vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/tools/ncurses/bin/ncursesw6-config"  "${CURRENT_INSTALLED_DIR}" "\${prefix}")
-vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/tools/ncurses/bin/ncursesw6-config"  "\nprefix=\"\${prefix}\"" [=[prefix=$(CDPATH= cd -- "$(dirname -- "$0")"/../../.. && pwd -P)]=])
-if(NOT VCPKG_BUILD_TYPE)
-    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/tools/ncurses/debug/bin/ncursesw6-config"  "${CURRENT_INSTALLED_DIR}" "\${prefix}")
-    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/tools/ncurses/debug/bin/ncursesw6-config"  "\nprefix=\"\${prefix}/debug\"" [=[prefix=$(CDPATH= cd -- "$(dirname -- "$0")"/../../../.. && pwd -P)/debug]=])
-    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/tools/ncurses/debug/bin/ncursesw6-config"  "\${prefix}/share" "\${prefix}/../share")
-endif()
-
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin")
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/bin")
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
@@ -91,5 +88,3 @@ file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
 
 file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
 vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/COPYING")
-
-
diff --git a/vcpkg.json b/vcpkg.json
index a9ba52c..9ec8abf 100644
--- a/vcpkg.json
+++ b/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "ncurses",
   "version": "6.5",
-  "port-version": 3,
+  "port-version": 1,
   "description": [
     "Free software emulation of curses in System V Release 4.0, and more.",
     "This port installs a wide character configuration (ncursesw)."
@@ -13,10 +13,6 @@
     {
       "name": "vcpkg-cmake-get-vars",
       "host": true
-    },
-    {
-      "name": "vcpkg-make",
-      "host": true
     }
   ],
   "features": {
